### PR TITLE
Switch to netty-tcnative 2.0.0 which uses different package names

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

Previous versions of netty-tcnative used the org.apache.tomcat namespace which could lead to problems when a user tried to use tomcat and netty in the same app.

Modifications:

Use netty-tcnative which now uses a different namespace and adjust code to some API changes.

Result:

Its now possible to use netty-tcnative even when running together with tomcat.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@Scottmitch @nmittler @trustin PTAL... this depends on https://github.com/netty/netty-tcnative/pull/182

**Comment by nmittler** [comment]
@normanmaurer LGTM ... looks like you managed to get rid of some of the APR gunk as well?

**Comment by normanmaurer** [comment]
The APR bits are now an implementation detail of tcnative.

File: handler/src/main/java/io/netty/handler/ssl/OpenSslSessionContext.java [hunk 1]
```diff
@@ -62,9 +63,21 @@ public SSLSession getSession(byte[] bytes) {
      */
     @Deprecated
     public void setTicketKeys(byte[] keys) {
-        ObjectUtil.checkNotNull(keys, "keys");
+        if (ObjectUtil.checkNotNull(keys, "keys").length % SessionTicketKey.TICKET_KEY_SIZE != 0) {
```
Comment (Scottmitch) [inline handler/src/main/java/io/netty/handler/ssl/OpenSslSessionContext.java]: `ObjectUtil.checkNotNull(keys, "keys")` -> `keys`. A NPE will be thrown anyways and `keys` is the only paramter.

**Comment by Scottmitch** [comment]
lgtm

**Comment by trustin** [comment]
LGTM

**Comment by Scottmitch** [comment]
@normanmaurer - Can we rebase and pull this in for the next release?

**Comment by normanmaurer** [comment]
Yes

**Comment by jchambers** [comment]
Hello! To follow up on the "next release" comment from earlier, is the thought that this might still happen for 4.1.7? I haven't quite been able to figure out where netty-tcnative 2.0.0 fits into the release plan.

Thanks kindly!

**Comment by normanmaurer** [comment]
@jchambers this will be 4.1.8.Final

**Comment by jchambers** [comment]
I hate to be a pest, but is this still on the to-do list for 4.1.8?

**Comment by normanmaurer** [comment]
@jchambers sorry but we will most likely delay it to 4.1.9.Final as 4.1.8.Final will be released within one week and we want to "keep" the risk low.

**Comment by jchambers** [comment]
Understood. Thanks for the update.

**Comment by marian09** [comment]
Hi, I'm getting this error triying to use pushy in my project who runs on tomcat 8:

2017-01-26 17:02:22 INFO  ApnsClientBuilder:424 - Native SSL provider not available; will use JDK SSL provider.
Exception in thread "ActiveMQ Session Task-1" java.lang.ExceptionInInitializerError
	at io.netty.handler.ssl.JdkSslContext.toNegotiator(JdkSslContext.java:283)
	at io.netty.handler.ssl.JdkSslClientContext.<init>(JdkSslClientContext.java:261)
	at io.netty.handler.ssl.SslContext.newClientContextInternal(SslContext.java:751)
	at io.netty.handler.ssl.SslContextBuilder.build(SslContextBuilder.java:406)
	¡at org.apache.activemq.ActiveMQMessageConsumer.dispatch(ActiveMQMessageConsumer.java:1321)
	at org.apache.activemq.ActiveMQSessionExecutor.dispatch(ActiveMQSessionExecutor.java:131)
	at org.apache.activemq.ActiveMQSessionExecutor.iterate(ActiveMQSessionExecutor.java:202)
	at org.apache.activemq.thread.PooledTaskRunner.runTask(PooledTaskRunner.java:129)
	at org.apache.activemq.thread.PooledTaskRunner$1.run(PooledTaskRunner.java:47)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.RuntimeException: ALPN unsupported. Is your classpatch configured correctly? See http://www.eclipse.org/jetty/documentation/current/alpn-chapter.html#alpn-starting
	at io.netty.handler.ssl.JdkAlpnApplicationProtocolNegotiator$1.<init>(JdkAlpnApplicationProtocolNegotiator.java:27)
	at io.netty.handler.ssl.JdkAlpnApplicationProtocolNegotiator.<clinit>(JdkAlpnApplicationProtocolNegotiator.java:24)
	... 16 more

I have those dependencies in my pom.xml file:

<dependency>
		<groupId>com.relayrides</groupId>
		<artifactId>pushy</artifactId>
		<version>0.9</version>
	</dependency>
	<dependency>
		<groupId>io.netty</groupId>
		<artifactId>netty-tcnative-boringssl-static</artifactId>
		<version>1.1.33.Fork24</version>
	</dependency>
	<dependency>
    	<groupId>io.netty</groupId>
    	<artifactId>netty-all</artifactId>
    	<version>4.1.6.Final</version>
		</dependency> 
	<dependency>
		<groupId>org.eclipse.jetty.alpn</groupId>
		<artifactId>alpn-api</artifactId>
		<version>1.1.3.v20160715</version>
	</dependency>

I don't know what I'm doing wrong, I put the jars in my lib folder too. Do I need to install a ssl provider in my computer? I hope you can help me. 
Thank you so much.

**Comment by Scottmitch** [comment]
@marian09 - Please open an issue if you think there is a Netty bug and see http://netty.io/wiki/forked-tomcat-native.html for more details on using netty-tcnative.

**Comment by jchambers** [comment]
@Scottmitch we're addressing this on the Pushy mailing list. @marian09 please do not open any new Netty bug reports for now.

**Comment by marian09** [comment]
Ok thank you and I'm sorry... I'm new on this haha

2017-01-26 17:42 GMT-04:00 Jon Chambers <notifications@github.com>:

**Comment by Scottmitch** [comment]
Thanks @jchambers. Also sorry again for the delay on this PR. We are eager to get this in ... it will happen!

File: handler/src/main/java/io/netty/handler/ssl/OpenSslSessionContext.java (hunk 1 above)
Comment (normanmaurer) [inline handler/src/main/java/io/netty/handler/ssl/OpenSslSessionContext.java]: sure...

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (1d128c7) and 4.0 (4109fe741f73f693c73d51853a082f2d00da803c)

**Pull request closed without merging**